packages/cli-repl/src/cli-repl.ts

```diff
@@ -263,10 +283,39 @@ class CliRepl {
       await snippetManager?.loadAllSnippets();
     }
     await this.loadRcFiles();
+    // We only enable/disable here, since the rc file/command line scripts
+    // can disable the telemetry setting.
+    this.setTelemetryEnabled(await this.getConfig('enableTelemetry'));
     this.bus.emit('mongosh:start-mongosh-repl', { version });
     await this.mongoshRepl.startRepl(initialized);
   }
 
+  setupAnalytics(): void {
+    if (process.env.IS_MONGOSH_EVERGREEN_CI && !this.analyticsOptions?.alwaysEnable) {
+      throw new Error('no analytics setup for the mongosh CI environment');
+    }
+    // build-info.json is created as a part of the release process
+    const apiKey = this.analyticsOptions?.apiKey ?? require('./build-info.json').segmentApiKey;
+    this.segmentAnalytics = new Analytics(
+      apiKey,
+      this.analyticsOptions);
+    this.toggleableAnalytics = new ToggleableAnalytics(this.segmentAnalytics);
+  }
+
+  setTelemetryEnabled(enabled: boolean): void {
+    if (this.globalConfig === null) {
```

Comment:
Maybe a bit nitpicky, but could you get here via user action? ie. a user wants to explicitly enable/disable telemetry and then it silently ignores that request.

Reply:
No, the global config is loaded pretty early – I think this line here would technically not even be necessary if the global config is loaded before the local one, but I’d also argue that this here is the safer behavior, because it prevents the user from enabling telemetry when that should not be allowed according to the global config